text/0474-event-bridge-scheduler-l2.md

```diff
@@ -0,0 +1,545 @@
+3. **Schedule Group**: Scheduling groups help you organize your schedules. Scheduling groups support tags that you can use for cost allocation, access control, and resource organization. When creating a new schedule, you can add it to a scheduling group.
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project. It allows you to define Event Bridge Schedules.
+
+
+## Defining a schedule 
+
+```ts
+
+const target = new targets.LambdaInvoke(props.func, {
+    input: ScheduleTargetInput.fromObject({
+        "payload": "useful"
+    })
+});
+    
+const schedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test schedule that invokes lambda function every 10 minutes.',
+});
+```
+
+### Schedule Expressions
+
+Both rate-based and cron-based schedules are recurring schedules. You configure each recurring schedule type using a schedule expression for the type of schedule you want to configure, and specifying a time zone in which EventBridge Scheduler evaluates the expression.
+
+```ts
+const rateBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test rate-based schedule',
+});
+
+const cronBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.cron({ day: '20', month: '11' }),
+    target,
+    scheduleTimeZone: TimeZone.AMERICA_NEW_YORK,
+    description: 'This is a test cron-based schedule in New York timezone',
+});
+```
+
+A one-time schedule is a schedule that invokes a target only once. You configure a one-time schedule when by specifying the time, date, and time zone in which EventBridge Scheduler evaluates the schedule.
+
+```ts
+const oneTimeSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.at(new Date(2022, 10, 20, 19, 20, 23)),
+    target,
+    scheduleTimeZone: TimeZone.AMERICA_NEW_YORK,
+    description: 'This is a one-time schedule in New York timezone',
+});
+```
+
+### Grouping Schedules
+
+When creating a new schedule, you can add it to a scheduling group:
+
+```ts
+const group = new Group(this, "Group", {
+    groupName: "MyGroup"
+});
+
+const target = new targets.LambdaInvoke(props.func, {
+    input: ScheduleTargetInput.fromObject({
+        "payload": "useful"
+    })
+});
+    
+const schedule1 = new Schedule(this, 'Schedule1', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target
+});
+
+const schedule2 = new Schedule(this, 'Schedule2', {
+    schedule: ScheduleExpression.rate(Duration.minutes(5)),
+    target
+});
+
+group.addSchedules(schedule1, schedule2);
+```
+
+If group is not provided schedule will be created in default group, which you can access in CDK with: 
+```ts
+const defaultGroup = Group.fromDefaultGroup(this, "DefaultGroup");
```

Comment:
Only for being able to do grantRead / grantWrite on the default group :-)

Reply:
> We'll have to worry about dependencies because we will use the defaultGroup resource as if it already exists and not as a to-be-created resource.

Yeah, as Jacco mentioned above the default group already exists in every AWS account. 

See https://docs.aws.amazon.com/scheduler/latest/UserGuide/managing-schedule-group.html

> Only for being able to do grantRead / grantWrite on the default group :-)

And also for getting its metrics.

Reply:
Sounds good. If it already exists, shouldn't be a dependency problem then.